variables.tf

```diff
@@ -318,10 +318,27 @@ variable "addons" {
     cluster-autoscaler        = optional(string)
     vpc-block-csi-driver      = optional(string)
     ibm-storage-operator      = optional(string)
+    openshift-ai              = optional(string)
   })
   description = "Map of OCP cluster add-on versions to install (NOTE: The 'vpc-block-csi-driver' add-on is installed by default for VPC clusters and 'ibm-storage-operator' is installed by default in OCP 4.15 and later, however you can explicitly specify it here if you wish to choose a later version than the default one). For full list of all supported add-ons and versions, see https://cloud.ibm.com/docs/containers?topic=containers-supported-cluster-addon-versions"
   nullable    = false
   default     = {}
+
+  validation {
+    condition     = lookup(var.addons, "openshift-ai", null) == null || (tonumber(local.ocp_version_num) >= 4.16)
+    error_message = "OCP AI add-on requires OCP version >= 4.16.0"
+  }
+
+  validation {
+    condition     = lookup(var.addons, "openshift-ai", null) == null || alltrue([for workers in values(local.workers_per_pool) : workers >= 2])
+    error_message = "OCP AI add-on requires at least 2 worker nodes in each worker pool."
+  }
+
+  validation {
+    condition     = lookup(var.addons, "openshift-ai", null) == null || alltrue([for spec in values(local.worker_specs) : spec.cpu_count >= 8 && spec.ram_count >= 32])
+    error_message = "To install OCP AI add-on, all worker nodes in all pools must have at least 8-core CPU and 32GB memory."
```

Comment:
I think there is still one big validation missing from this one, from the official add-on docs:
> Each worker node must have 8vCPU and 32GB memory and a GPU

That last part is I think the most important, it must be GPU machines. I think those are noted by starting with a "g" as first letter. Looking in the UI, I think the smallest GPU for openshift is "gx3.16x80.l4"

Reply:
Yes that is correct about smallest GPU. Hence gave that as default value.
Now added validation to include GPU machine.